api/speciesnet/Dockerfile.sagemaker

```diff
@@ -0,0 +1,47 @@
+# Custom Docker image to run speciesnet in SageMaker
+
+FROM public.ecr.aws/docker/library/python:3.11-slim
+
+# Install system packages and clean up in same layer
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    pkg-config \
+    libhdf5-dev \
+    libgl1-mesa-glx \
+    libglib2.0-0 \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Python packages efficiently
+COPY requirements.txt /opt/ml/requirements.txt
+RUN pip install --no-cache-dir -r /opt/ml/requirements.txt \
+    && pip install --no-cache-dir 'speciesnet[server]' kagglehub \
+    && find /usr/local/lib/python3.11/site-packages -name "*.pyc" -delete
+
+# Create directories required by SageMaker
+RUN mkdir -p /opt/ml/{model,input,output}
+
+# Set KaggleHub cache directory
+RUN mkdir -p /opt/ml/model/speciesnet
+ENV KAGGLEHUB_CACHE=/opt/ml/model/speciesnet
+
+# Download model during build
+RUN python -c "import kagglehub; kagglehub.model_download('google/speciesnet/keras/v4.0.0a')"
```

Comment:
Download the model artifacts during build time. Was hitting the 180s hard timeout on sagemaker while loading the model during runtime.